cmd/ketch/app_deploy.go

```diff
@@ -111,20 +152,59 @@ func appDeploy(ctx context.Context, cfg config, getImageConfigFile getImageConfi
 		}
 		exposedPorts = append(exposedPorts, *exposedPort)
 	}
+
+	// default deployment spec for an app
 	deploymentSpec := ketchv1.AppDeploymentSpec{
 		Image:     options.image,
 		Version:   ketchv1.DeploymentVersion(app.Spec.DeploymentsCount + 1),
 		Processes: processes,
 		KetchYaml: ketchYaml,
 		RoutingSettings: ketchv1.RoutingSettings{
-			Weight: 100,
+			Weight: defaultTrafficWeight,
 		},
 		ExposedPorts: exposedPorts,
 	}
-	app.Spec.Deployments = []ketchv1.AppDeploymentSpec{
-		deploymentSpec,
+
+	if options.isCanarySet() {
+		if err := options.validateCanaryOpts(); err != nil {
+			return err
+		}
+
+		// check for old deployements
+		switch deps := len(app.Spec.Deployments); {
+		case deps == 0:
+			return fmt.Errorf("Canary deployment failed. No primary deployment found for the app")
+		case deps >= 2:
+			return fmt.Errorf("Canary deployment failed. Maximum number of two deployments are currently supported")
+		}
+
+		// parses step interval string to time.Duration
+		stepInt, _ := time.ParseDuration(options.stepTimeInterval)
+		nextSch := metav1.NewTime(timeNow().Add(stepInt))
+		app.Spec.Canary = ketchv1.CanarySpec{
+			Steps:             options.steps,
+			StepWeight:        options.stepWeight,
+			StepTimeInteval:   stepInt,
+			NextScheduledTime: &nextSch,
+		}
+
+		app.Status.CurrentCanaryStep = 1
+		app.Status.IsActiveCanary = true
+
+		// set weight for canary deployment
+		deploymentSpec.RoutingSettings.Weight = options.stepWeight
+
+		//  update old deployment weight
+		app.Spec.Deployments[0].RoutingSettings.Weight = defaultTrafficWeight - options.stepWeight
```

Comment:
So like shipa, we only support 2 deployments. One is primary and the second is canary ( if required by the user).
We are not ignoring other deployments as we set it [here](https://github.com/shipa-corp/ketch/pull/37/files#diff-f63ce58d17c4d38db0ecd7d4ac149e842151eefe4ff3692beeb7c41f934d112dR187)

and we only allow canary in case a primary deployment exists https://github.com/shipa-corp/ketch/pull/37/files#diff-f63ce58d17c4d38db0ecd7d4ac149e842151eefe4ff3692beeb7c41f934d112dR176